src/pycram/local_transformer.py

```diff
@@ -93,9 +107,13 @@ def lookup_transform_from_source_to_target_frame(self, source_frame: str, target
         Update the transforms for all world objects then Look up for the latest known transform that transforms a point
          from source frame to target frame. If no time is given the last common time between the two frames is used.
 
+        :param source_frame: The frame in which the point is currently represented
+        :param target_frame: The frame in which the point should be represented
         :param time: Time at which the transform should be looked up
+        :return: The transform from source_frame to target_frame
         """
-        self.world.update_transforms_for_objects_in_current_world()
+        self.update_transforms_for_objects(source_frame.split("/")[0], target_frame.split("/")[0])
```

Comment:
this name split should be a function. Consider that other people are doing something similar and are wondering how to extract the relevatn part. Perhaps they do it slightly wrong in their own way. A common function for it ensures long lasting support